optima/utils.py

```diff
@@ -894,7 +900,24 @@ def smoothinterp(newx=None, origx=None, origy=None, smoothness=None, growth=None
         if len(futureindices): # If there are past data points
             lastpoint = futureindices[0]-1
             newy[futureindices] = newy[lastpoint] * exp((newx[futureindices]-newx[lastpoint])*growth) # Get last 'good' data point and apply growth
-        
+    
+    # Add infinities back in, if they exist
+    if any(~isfinite(origy)): # Infinities exist, need to add them back in manually since interp can only handle nan
+        if not keepfinite: # Don't keep it finite
+            orignan      = zeros(len(origy)) # Start from scratch
+            origplusinf  = zeros(len(origy)) # Start from scratch
+            origminusinf = zeros(len(origy)) # Start from scratch
+            orignan[isnan(origy)]     = nan  # Replace nan entries with nan
+            origplusinf[origy==inf]   = nan  # Replace plus infinite entries with nan
+            origminusinf[origy==-inf] = nan  # Replace minus infinite entries with nan
+            newnan      = interp(newx, origx, orignan) # Interpolate the nans
+            newplusinf  = interp(newx, origx, origplusinf) # ...again, for positive
+            newminusinf = interp(newx, origx, origminusinf) # ...and again, for negative
+            newy[isnan(newminusinf)] = -inf # Add minus infinity back in first
+            newy[isnan(newplusinf)]  = inf # Then, plus infinity
+            newy[isnan(newnan)]  = nan # Finally, the nans
```

Comment:
allows interp to handle infinite values (fundamental bug in all of python's interp algorithms, afaics)